app/src/androidTest/java/ch/epfl/sweng/runpharaa/FavoritesFragmentTest.java

```diff
@@ -0,0 +1,127 @@
+import android.support.test.espresso.Espresso;
+import android.support.test.espresso.UiController;
+import android.support.test.espresso.ViewAction;
+import android.support.test.rule.ActivityTestRule;
+import android.view.View;
+
+import com.google.android.gms.maps.model.LatLng;
+
+import org.hamcrest.Matcher;
+import org.junit.BeforeClass;
+import org.junit.Ignore;
+import org.junit.Rule;
+import org.junit.Test;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import ch.epfl.sweng.runpharaa.Initializer.TestInitLocation;
+import ch.epfl.sweng.runpharaa.tracks.Track;
+import ch.epfl.sweng.runpharaa.user.User;
+
+import static android.os.SystemClock.sleep;
+import static android.support.test.espresso.Espresso.onView;
+import static android.support.test.espresso.action.ViewActions.click;
+import static android.support.test.espresso.action.ViewActions.swipeDown;
+import static android.support.test.espresso.action.ViewActions.swipeLeft;
+import static android.support.test.espresso.assertion.ViewAssertions.matches;
+import static android.support.test.espresso.contrib.RecyclerViewActions.actionOnItemAtPosition;
+import static android.support.test.espresso.matcher.ViewMatchers.isDisplayed;
+import static android.support.test.espresso.matcher.ViewMatchers.isDisplayingAtLeast;
+import static android.support.test.espresso.matcher.ViewMatchers.withId;
+import static android.support.test.espresso.matcher.ViewMatchers.withText;
+import static android.support.test.internal.runner.junit4.statement.UiThreadStatement.runOnUiThread;
+import static org.hamcrest.core.AllOf.allOf;
+
+public class FavoritesFragmentTest extends TestInitLocation {
+    @Rule
+    public final ActivityTestRule<MainActivity> mActivityRule =
+            new ActivityTestRule<>(MainActivity.class);
+
+    @BeforeClass
+    public static void initUser() {
+        User.set("FakeUser", 2000, Uri.parse(""), new LatLng(37.422, -122.084), "1");
+
+    }
+
+    private Track t;
+
+    @Test
+    public void testNoFavorites() throws Throwable {
+        Context targetContext = InstrumentationRegistry.getInstrumentation()
+                .getTargetContext();
+
+        onView(withId(R.id.viewPagerId)).perform(swipeLeft());
+        onView(withId(R.id.viewPagerId)).perform(swipeLeft());
+        sleep(2000);
+
+        //refresh
+        runOnUiThread(() ->((FragmentFavourites)mActivityRule.getActivity().getSupportFragmentManager().getFragments().get(2)).onRefresh());
+
+        sleep(1000);
+
+        onView(allOf(withId(R.id.emptyMessage), isDisplayed())).check(matches(withText(targetContext.getResources().getString(R.string.no_favorite))));
+    }
+
+    @Test
+    public void testFavoritesAppears() throws Throwable {
+        sleep(2000);
+        runOnUiThread(() ->((FragmentNearMe)mActivityRule.getActivity().getSupportFragmentManager().getFragments().get(0)).onRefresh());
+        /*onView(allOf(withId(R.id.cardListId), isDisplayingAtLeast(75))).perform(
```

Comment:
Just make sure to delete the comments if it does not work

Reply:
I removed them in the latest commit, thanks for pointing it out.